### DISCUSSION TITLE
How to have a modal action fired from a custom button inside a page?

### DISCUSSION THREAD

**Discussion by maxacarvalho** [filamentphp/filament#6687] [Help] [answered]
Hi,

Is it possible to fire an action with a modal from a custom button inside a page?

I have a custom blade edit page for a resource. Inside that blade file, I have a custom button.
That button triggers a custom method inside the resource EditPage. I'd like to have that method to start a modal action.

Here's the blade snippet:

```
<x-filament-support::button
    color="danger"
    :dark-mode="config('filament.dark_mode')"
    tag="button"
    type="button"
    wire:click="customMethod"
>
    Some label...
</x-filament-support::button>
```

And here's the custom method declared inside the resource's EditPage
```php
public function customMethod(): void
{
    ray('sendQuote'); // this works fine
    Action::make('something')->action(fn () => doSomething())->requiresConfirmation(); // this doesn't do anything
}
```

Thank you.

**Comment by danharrin** [comment, marked as answer]
In v3 you will be able to do it, but at the moment you have to use `getActions()`
